receiver/splunksearchapireceiver/receiver.go

```diff
@@ -67,66 +67,83 @@ func (ssapir *splunksearchapireceiver) runQueries(ctx context.Context) error {
 			ssapir.logger.Error("error polling for search completion", zap.Error(err))
 		}
 
-		// fetch search results
-		ssapir.logger.Info("fetching search results")
-		results, err := ssapir.getSplunkSearchResults(searchID)
-		if err != nil {
-			ssapir.logger.Error("error fetching search results", zap.Error(err))
-		}
-		ssapir.logger.Info("search results fetched", zap.Int("num_results", len(results.Results)))
+		var resultCountTracker = 0 // track number of results exported
+		var offset = 0             // offset for pagination
+		var limitReached = false
+		for {
```

Comment:
Let's set up some time to vscode share a solution on polling; feel like having an infinite for loop in this case could actually be improved if we start looking at context.Context